datumaro/plugins/yolo_format/extractor.py

```diff
@@ -415,11 +415,23 @@ def _load_categories(self) -> CategoriesInfo:
         return {AnnotationType.label: LabelCategories.from_iterable(names)}
 
     def _get_labels_path_from_image_path(self, image_path: str) -> str:
-        relative_image_path = osp.relpath(
-            image_path, osp.join(self._path, YOLOv8Path.IMAGES_FOLDER_NAME)
-        )
-        relative_labels_path = osp.splitext(relative_image_path)[0] + YOLOv8Path.LABELS_EXT
-        return osp.join(self._path, YOLOv8Path.LABELS_FOLDER_NAME, relative_labels_path)
+        rel_image_path = osp.relpath(image_path, self._path)
+        split_rel_path = rel_image_path.split(osp.sep, 2)
+        if len(split_rel_path) < 3 or not any(
+            pc == YOLOv8Path.IMAGES_FOLDER_NAME for pc in split_rel_path[:2]
+        ):
+            raise InvalidAnnotationError(f"Malformed folder structure for image {rel_image_path}")
+
+        split_rel_path[
+            next(
+                index
+                for index in range(len(split_rel_path) - 1, -1, -1)
+                if split_rel_path[index] == YOLOv8Path.IMAGES_FOLDER_NAME
+            )
+        ] = YOLOv8Path.LABELS_FOLDER_NAME
```

Comment:
Is it possible that both elements are `YOLOv8Path.IMAGES_FOLDER_NAME`? If so, maybe it makes sense to check if there is a file before returning the result?

Reply:
In case both elements are "images" I think we should behave the same way as ultralytics behaves - replace the last one: https://github.com/ultralytics/ultralytics/blob/main/ultralytics/data/utils.py#L44